app/models/comment.rb

```diff
@@ -0,0 +1,30 @@
+class Comment < ActiveRecord::Base
+  belongs_to :article
+  attr_accessible :content, :email, :subject
+  validates :subject, presence: true
+  validates :email, presence: true
+  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+
+  def self.cached_article(id)
+    Rails.cache.fetch(["#{self.name}_article", id]) do
+      find(id).article
+    end
+  end
+
+  def self.cached_find(id)
+    Rails.cache.fetch([self.name, id]) do
+      find(id)
+    end
+  end
+
+  def self.cached_all(article_id)
+    Rails.cache.fetch([self.name, 'all_#{article_id}']) do
+      result = all.select{|comment| comment.article_id = article_id}
```

Comment:
@phendych this is wrong, but as a sidenote, this .select is an array operation. so if you have 1 million comments, it will take all 1 million rows from DB and then do a filtering on the array, not efficient at all.